src/Request.ts

```diff
@@ -0,0 +1,320 @@
+	// 		in the query string.
+	selectParam: string;
+
+	// sortParam: String
+	//		The query parameter to used for holding sort information. If this is omitted, than
+	//		the sort information is included in a functional query token to avoid colliding
+	//		with the set of name/value pairs.
+	sortParam: string;
+
+	// target: String
+	//		The target base URL to use for all requests to the server. This string will be
+	//		prepended to the id to generate the URL (relative or absolute) for requests
+	//		sent to the server
+	target: string;
+
+	// useRangeHeaders: Boolean
+	//		The indicates if range limits (start and end) should be specified
+	//		a Range header, using items units. If this is set to true, a header
+	//		be included of the form:
+	//			Range: items=start-end
+	useRangeHeaders: boolean;
+
+	// summary:
+	//		This is a basic store for RESTful communicating with a server through JSON
+	//		formatted data. It extends dstore/Store.
+	constructor(options?: RequestStoreArgs) {
+		super(options);
+		// summary:
+		//		This is a basic store for RESTful communicating with a server through JSON
+		//		formatted data.
+		// options: dstore/JsonRest
+		//		This provides any configuration information that will be mixed into the store
+
+		//	Defaults to JSON, but other formats can be parsed by providing an alternate
+		//	parsing function. If you do want to use an alternate format, you will probably
+		//	want to use an alternate stringify function for the serialization of data as well.
+		//	Also, if you want to support parsing a larger set of JavaScript objects
+		//	outside of strict JSON parsing, you can provide dojo/_base/json.fromJson as the parse function
+		this.parse = this.parse || function (data: string) {
+			return JSON.parse(data);
+		};
+		this.headers || (this.headers = {});
+		this.target = this.target || '';
+		this._targetContainsQueryString = this.target.lastIndexOf('?') >= 0;
+		this.ascendingPrefix = this.ascendingPrefix || '+';
+		this.descendingPrefix = this.descendingPrefix || '-';
+		this.accepts = this.accepts || 'application/json';
+		this.fetch = this.fetch || function (kwArgs?: dstore.FetchArgs) {
+			var results = this._request(kwArgs);
+			return <dstore.FetchPromise<any>>QueryResults(results.data, {
+				response: results.response
+			});
+		}
+	}
+
+	fetchRange(kwArgs: dstore.FetchRangeArgs) {
+		var start = kwArgs.start,
+			end = kwArgs.end,
+			requestArgs: dstore.FetchArgs = {};
+		if (this.useRangeHeaders) {
+			requestArgs.headers = lang.mixin(this._renderRangeHeaders(start, end), kwArgs.headers);
+		} else {
+			requestArgs.queryParams = this._renderRangeParams(start, end);
+			if (kwArgs.headers) {
+				requestArgs.headers = kwArgs.headers;
+			}
+		}
+
+		var results = this._request(requestArgs);
+		return <dstore.FetchPromise<T>>QueryResults(results.data, {
+			totalLength: results.total,
+			response: results.response
+		});
+	}
+
+	_request(kwArgs: dstore.FetchArgs) {
+		kwArgs = kwArgs || {};
```

Comment:
you should be able to default this in the arguments declaration: `(kwArgs: dstore.FetchArgs = {})`